persistence-api/src/main/java/io/stargate/db/query/BoundDMLQueryWithConditions.java

```diff
@@ -0,0 +1,13 @@
+package io.stargate.db.query;
+
+import java.util.List;
+
+public interface BoundDMLQueryWithConditions extends BoundDMLQuery {
```

Comment:
`BoundInsert` extends `BoundDMLQuery` but not `BoundDMLQueryWithConditions`. `BoundDMLQueryWithConditions` is mostly to save duplication between `BoundUpdate` and `BoundDelete` and I expect it to be used rarely otherwise (but, well, it can also to save duplication in the handling of `BoundUpdate` and `BoundDelete` as well obviously).